practical/matlab/idr0021_steps.m

```diff
@@ -0,0 +1,203 @@
+    for j = 1 : numel(channels) 
+        channel = channels(j);
+        channelId = channel.getId().getValue();
+        lc = channel.getLogicalChannel();
+        channelName = lc.getName().getValue();
+        if channelName == target
+            channelIndex = j;
+            disp(channelIndex);
+        end
+    end
+end
+
+
+% Exercise 5
+% Perform image segmentation on one image.
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+fprintf('Using image: %s, channel: %s (index: %i)\n', image.getName().getValue(), target, channelIndex)
+z = 0;
+t = 0;
+plane = getPlane(session, image, z, channelIndex - 1, t);
+threshNstd = 6;
+minPixelsPerCentriole = 20;   % minimum size of objects of interest
+vals = reshape(plane, [numel(plane), 1]);   % reshape to 1 column
+mean1 = mean(vals);
+std1 = std(vals);
+% images are mostly background, so estimate threshold using basic stats
+thresh1 = mean1 + threshNstd * std1;
+bwRaw = imbinarize(plane, thresh1);
+BWfinal = bwareaopen(bwRaw, minPixelsPerCentriole);  % remove small objects
+imshow(BWfinal);
+title(strcat(string(image.getName().getValue()), ' (segmented)'));
+
+
+% Exercise 6 
+% Putting it all together: Analyse the whole dataset and
+% save results as ROIs and CSV file.
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+iUpdate = session.getUpdateService(); % needed to save the ROIs
+csv_data = java.util.ArrayList;
+for i = 1 : length(datasetImages)
+    image = datasetImages(i);
+    channels = loadChannels(session, image);
+    for j = 1 : numel(channels) 
+        channel = channels(j);
+        channelId = channel.getId().getValue();
+        lc = channel.getLogicalChannel();
+        channelName = lc.getName().getValue();
+        if channelName == target
+            channelIndex = j;
+            break;
+        end
+    end
+    
+    fprintf('Analyse Image: %s, Channel: %i\n', image.getName().getValue(), channelIndex);
+    plane = getPlane(session, image, z, channelIndex - 1, t); % channel index in OMERO starts with 0
+    threshNstd = 6;
+    minPixelsPerCentriole = 20;   % minimum size of objects of interest
+    vals = reshape(plane, [numel(plane), 1]);   % reshape to 1 column
+    mean1 = mean(vals);
+    std1 = std(vals);
+    % images are mostly background, so estimate threshold using basic stats
+    thresh1 = mean1 + threshNstd * std1;
+    bwRaw = imbinarize(plane, thresh1);
+    BWfinal = bwareaopen(bwRaw, minPixelsPerCentriole);
+    
+    [B,L] = bwboundaries(BWfinal, 'noholes');
+    roi = omero.model.RoiI;
+    max_area = 0;
+    csv_row = java.util.ArrayList;
+    for b = 1:length(B)
+        boundary = B{b};
+        x_coordinates = boundary(:,2);
+        y_coordinates = boundary(:,1);
+        shape = createPolygon(x_coordinates, y_coordinates);
+        setShapeCoordinates(shape, z, channelIndex - 1, t);
+        roi.addShape(shape);
+        area = polyarea(x_coordinates, y_coordinates);
+        max_area = max(max_area, area);
```

Comment:
could we maybe harvest more parameters from matlab into the CSV ? the result table looks a bit meagre

Reply:
As discussed, any intensity values would be great.